### DISCUSSION TITLE
Issues with quasi-crystal mdf

### DISCUSSION THREAD

**Discussion by ashwinShahani** [mtex-toolbox/mtex#928] [Ask Anything] [answered]
Hello,

I am interested in plotting the disorientation angle distribution for a uniform ODF of a decagonal quasicrystal (non-crystallographic point group 10 2 2).  I execute the below code

```
rot10 = rotation.byAxisAngle(vector3d.Z,36*degree);
rot2x = rotation.byAxisAngle(vector3d.X,180*degree);
rot2y = rotation.byAxisAngle(vector3d.Y,180*degree);
cs = crystalSymmetry.byElements([rot10,rot2x,rot2y]);
figure, plotAngleDistribution(uniformODF(cs,cs))
```

and observe in the angle distribution a "sawtooth" pattern that is amplified at high disorientation angles, see below image.  What is the origin of this pattern?  Is it a computational artifact?   Note that it does not manifest when I change the crystal symmetry to something more conventional (like 432), so I suspect the underlying problem may be with how I declare the decagonal crystal symmetry at the onset.  

![sawtooth](https://user-images.githubusercontent.com/81767898/113326088-3ccb7700-92e7-11eb-91b5-5bcb04221755.png)

Secondly, when I check the MDF (plotted as sections of angle-axis space), 

```
mdf = calcMDF(uniformODF(cs));
figure, plotSection(mdf,'axisAngle')
```

I run into the following error with plotSection(): 

```
Error using nfsoftmex
Input argument N must be positive.

Error in FourierComponent/eval (line 41)
  plan =
  nfsoftmex('init',L,length(ori),nfsoft_flags,0,4,1000,2*ceil(1.5*L));

Error in ODF/eval (line 30)
    eval(odf.components{i},g,varargin{:}),size(g));

Error in ODF/plotSection (line 29)
Z = odf.eval(S3G);
```

Any advice on how to resolve these above two issues would be gratefully appreciated.

Thanks,
Ashwin

**Comment by ralfHielscher** [comment, marked as answer]
Hi Ashwin,

1. yes this is a numerical artifact. You can reduce it by changing in `orientationRegion/calcAngleDistribution` line 27 the resolution from `0.5*degree` to `0.05*degree`
2. yes this seems to be a bug. You can work around this by setting `mdf = uniformODF(cs,cs)`

Ralf.

**Reply by ashwinShahani** [reply]
This worked.  Thanks!
